### PR TITLE
Handle markers on same location

### PR DESCRIPTION
**Pull request by Morantron**
Addressing https://github.com/openfoodfoundation/openfoodnetwork/issues/1782

#### What? Why?

Map section cannot handle properly markers on same location.

#### What should we test?

Try map section, see if marker clustering feels good.

#### Release notes

Enterprises that share location now properly show in the map.

### CODE REVIEW

**Comment by daniellemoorhead** [comment]
Hey @Morantron do you think you could fill in the PR template with details? This helps a lot in the code review and testing process. Cheers!

**Comment by daniellemoorhead** [comment]
@sstead it would be great if we could get you to test this from a user's perspective before we push it live. And @mkllnk could you please code review this.

**Comment by sauloperez** [comment]
@Morantron anything left for us to start reviewing? can we remove the WIP already?

**Comment by Morantron** [comment]
Hi! I've pushed latest changes, this is not WIP anymore.

Attaching an screencast showing the new behavior.

[Screencast](https://imgur.com/a/OVkIQ)

File: app/assets/javascripts/darkswarm/services/enterprise_modal.js.coffee [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
-    open: (enterprise)->
+    open: (enterprises)->
       scope = $rootScope.$new(true) # Spawn an isolate to contain the enterprise
+      scope.enterprises = if angular.isArray(enterprises)
+        enterprises
+      else
+        [enterprises]
 
-      scope.enterprise = enterprise
+      console.log(scope.enterprises)
```
Comment (sauloperez) [inline app/assets/javascripts/darkswarm/services/enterprise_modal.js.coffee]: :warning:

File: config/database.yml [hunk 2]
```diff
@@ -22,13 +22,13 @@ production:
...
+  username: ofn_user
   password: f00d
 
 staging:
   adapter: postgresql
   encoding: unicode
   database: open_food_network_prod
   pool: 5
-  username: ofn
+  username: ofn_user
```
Comment (sauloperez) [inline config/database.yml]: :warning:

**Comment by sauloperez** [comment]
I'm very sorry @Morantron but there's a spec failing in https://travis-ci.org/openfoodfoundation/openfoodnetwork/jobs/283757904

**Comment by sstead** [comment]
This is a good first visualisation of what this could look like, but might need some more thought...

1) When a user opens the map, we want the level of zoom to be just enough to encompass all enterprises on the map. A comparison between how the map looks when first opened on Production and Staging is shown below for Australia... On staging it's zoomed in on a random point, which is not ideal.
image

2) Further, when I zoom out, there appears far fewer enterprises than before they were clustered. This makes the map look a bit sparse. I'm not sure what the best solution is, because as you can see above our production looks very cluttered, but we also don't want the map to look empty.
image

3) Clicking on this marker does nothing. The system needs to zoom in another layer, when this 3 splits into 1 and 2 markers. It's not zooming upon clicking- I need to manually zoom to see the profiles.
image
Sometimes this also happens at a more zoomed out level...
image

4) When >1 profiles are at the same address their profiles open as one modal, and you can scroll through each profile. This is a problem for some of our users such as farmer's markets of hubs, who create many (30+) profiles at the same address.
image

5) We're losing the red/green categories. For example in the pic below, 3 of the profiles are producers and one is a hub, yet the marker is red. The user can't tell if they are producer or not at a glance.
image
And when you look at the map broadly there is also few green markers..
image

6) This feature hasn't been extended to the group maps. Eventually it makes sense that both maps are consistent.
image

**Comment by sauloperez** [comment]
Thanks @sstead this clearly needs a second thought. We thought this could fix #1782 but it has more implications than expected. Let's close until we come up with a better solution that considers all your findings.

**Comment by sigmundpetersen** [comment]
Should it be taken out of Oct milestone @sauloperez if it's not going to be merged?

**Comment by sauloperez** [comment]
Oops, sorry. Thanks for the heads-up @sigmundpetersen . We won't be touching this again as we have other and more prioritary issues.

**Pull request closed without merging**